tests/spyre_util.py

```diff
@@ -1,18 +1,144 @@
 import math
 import os
-from typing import Any, Dict, List, Tuple, Union
+import subprocess
+import sys
+import time
+from typing import Any, Dict, List, Optional, Tuple, Union
 
 import numpy as np
+import openai
+import requests
 from sentence_transformers import SentenceTransformer, util
 from transformers import AutoModelForCausalLM, AutoTokenizer
 from vllm import LLM, SamplingParams
+from vllm.entrypoints.openai.cli_args import make_arg_parser
+from vllm.utils import FlexibleArgumentParser, get_open_port
 
 DISABLE_ASSERTS = False  # used for debugging
 
 ISCLOSE_REL_TOL_CPU = 0.1
 ISCLOSE_REL_TOL_SPYRE = 0.1
 
 
+class RemoteOpenAIServer:
+    """Subprocess wrapper that boots a vllm server with `vllm serve` for testing
+    against"""
+
+    DUMMY_API_KEY = "token-abc123"  # vLLM's OpenAI server does not need API key
+
+    def __init__(self,
+                 model: str,
+                 vllm_serve_args: list[str],
+                 *,
+                 env_dict: Optional[dict[str, str]] = None,
+                 seed: Optional[int] = 0,
+                 auto_port: bool = True,
+                 max_wait_seconds: Optional[float] = None,
+                 tensor_parallel_size: Optional[int] = 1) -> None:
```

Comment:
Since this is only used to set args in `vllm_serve_args`, can we instead call this like
```
with RemoteOpenAIServer(
            model, ["--tensor-parallel-size", "1"], ...)
```
?

Reply:
I updated the util and the test so that it would be called like you said but still through a pytest parameter.